### PR TITLE
Ports: Upgrade to FFmpeg 5.0 and make it run well

### PR DESCRIPTION
**Pull request by rouseabout**
Previously the FFmpeg port would build, but did not actually work (!).

To test inside SerenityOS type `ffplay ~/Videos/test-webm.webm` into the terminal.

### CODE REVIEW

File: Userland/Libraries/LibC/mallocdefs.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ struct BigAllocationBlock : public CommonHeader {
         m_magic = MAGIC_BIGALLOC_HEADER;
         m_size = size;
     }
-    unsigned char* m_slot[0];
+    [[gnu::aligned(16)]] unsigned char* m_slot[0];
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibC/mallocdefs.h:49]: Please just use standard `alignas(16)`.

File: AK/PrintfImplementation.h [hunk 2]
```diff
@@ -165,7 +166,26 @@ ALWAYS_INLINE int print_double(PutChFunc putch, CharType*& bufptr, double number
...
+        }
+        if (sign) {
+            putch(bufptr, '-');
+            length++;
+        }
+        if (nan) {
+            putch(bufptr, 'n'); putch(bufptr, 'a'); putch(bufptr, 'n');
+        } else {
+            putch(bufptr, 'i'); putch(bufptr, 'n'); putch(bufptr, 'f');
+        }
```
Comment (kleinesfilmroellchen) [inline AK/PrintfImplementation.h:186]: For these single line if's, we can get rid of the braces.

File: Ports/ffmpeg/package.sh [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+files="https://ffmpeg.org/releases/ffmpeg-${version}.tar.gz ffmpeg-${version}.tar.gz 7bf52bc242b5db8df67c62cb826df134d917dedcf6abf1289e15e4057bcc1750"
 auth_type="sha256"
 installopts=("INSTALL_TOP=${SERENITY_INSTALL_ROOT}/usr/local")
 configopts=("SRC_PATH=.")
 
 configure() {
     run ./configure \
-        --arch="x86_32" \
+        --target-os=none \
+        --arch="${SERENITY_ARCH}" \
```
Comment (kleinesfilmroellchen) [inline Ports/ffmpeg/package.sh:14]: Does this work even though our 32-bit arch is `i686` and not `x86_32` as the configure script received before?

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]

**Comment by kleinesfilmroellchen** [comment]
Great work! I've tested with the standard test video as well as FLAC, WAV and MP3, and they all work great including the spectral display.

**Comment by paulwratt** [comment]
hmm.. successfully tested - one step away from live streaming the desktop to Twitch ..

File: Userland/Libraries/LibC/inttypes.h [hunk 4]
```diff
@@ -41,10 +41,10 @@ __BEGIN_DECLS
...
 #    define PRIuPTR "lx"
 #endif
-#define PRIx8 "b"
-#define PRIX8 "hhX"
-#define PRIx16 "w"
-#define PRIX16 "hX"
+#define PRIx8 "x"
+#define PRIX8 "X"
+#define PRIx16 "x"
+#define PRIX16 "X"
```
[the remark this answers is not in the document]
Comment (awesomekling) [reply Userland/Libraries/LibC/inttypes.h:47]: Our `printf` used to support `%b` and `%w` as shorthand for 8-bit and 16-bit values. That's where it came from :^)

File: Ports/ffmpeg/patches/errno-is-not-valid-during-cpp.patch [hunk 5]
```diff
@@ -0,0 +1,13 @@
+diff --git a/libavutil/error.h b/libavutil/error.h
+index 0d3269a..a54d80d 100644
+--- a/libavutil/error.h
++++ b/libavutil/error.h
+@@ -37,7 +37,7 @@
+ 
+ 
+ /* error handling */
+-#if EDOM > 0
```
Comment (awesomekling) [inline Ports/ffmpeg/patches/errno-is-not-valid-during-cpp.patch:9]: Oh my, that is an unfortunate construct..

**Pull request merged**